bugsbunnyctf2017-rev150/solve.py

```diff
@@ -0,0 +1,61 @@
+from manticore.native import Manticore
+import sys
+
+prog = sys.argv[1]
+params = sys.argv[2:]
+
+"""
+To run:
+$ python rev150.py ./rev150 00000000000000000000
+
+The symbolic execution will overwrite the integer value with some number from 0 to 99999999999999999999 (you know that there are only 20 digits from IDA Pro. The program increments until the correct value is found.
+
+For testing purposes, the solution is 42813724579039578812.
+
+Run time: 9:10.62
+"""
+
+m = Manticore(prog, params)
+
+"""
+Here, we are setting the initial password (from IDA Pro, we can deduce a majority of the digits). We know that the password consists of 20 digits, all of which must be numeric.
+"""
+with m.locked_context() as context: 
+    context['password'] = 42810720579039578812
+
+@m.hook(0x401be2)
+def inject_password(state):
+    """
+    At this point, we inject our chosen password into the address holding the password inputted to the
+    binary. The password is formatted to be 20 digits long.
+    """
+    with m.locked_context() as context:
+        print("[+] injecting password: " + str(format(context['password'], '020')))
+        state.cpu.write_bytes(state.cpu.RDI,str(format(context['password'],'020')))
```

Comment:
The `format` function returns a string so we don't need to use `str`. Also, since we use the value twice, can  we assign it to a variable? e.g. `formatted_pwd = f"{context['password']:020}"`?